File: 1109. Corporate Flight Bookings/1109.go
package leetcode

func corpFlightBookings(bookings [][]int, n int) []int {
	res := make([]int,  n+1)
	for i:=0; i<len(bookings); i++{
		res[bookings[i][0]-1] += bookings[i][2]
		res[bookings[i][1]] -= bookings[i][2] 
	}
	for i:=1; i<n; i++{
		res[i] += res[i-1]
	}
	return res[:n]
}

File: 1109. Corporate Flight Bookings/_1109.py
class Solution:
    def corpFlightBookings(self, bookings, n: int) :
        res = [0]*(n+1)
        for item in bookings:
            res[item[0]-1] += item[2]
            res[item[1]] -= item[2]
        for i in range(1,n):
            res[i] += res[i-1] 
        return res[:-1]
